module/VuFind/src/VuFind/Search/Blender/Params.php

```diff
@@ -0,0 +1,655 @@
+
+            $result->set(
+                "query_$backendId",
+                $params->getQuery()
+            );
+            $result->set(
+                "params_$backendId",
+                $params->getBackendParameters()
+            );
+        }
+        return $result;
+    }
+
+    /**
+     * Add default filters to the given params
+     *
+     * @param BaseParams $params    Params
+     * @param string     $backendId Backend ID
+     *
+     * @return void
+     */
+    protected function addDefaultFilters(BaseParams $params, string $backendId): void
+    {
+        foreach ($this->mappings['Facets']['Fields'] ?? [] as $fieldConfig) {
+            $mappings = $fieldConfig['Mappings'][$backendId] ?? [];
+            $defaultValue = $mappings['DefaultValue'] ?? null;
+            if (null !== $defaultValue) {
+                $translatedField = $mappings['Field'];
+                $filterList = $params->getFilterList();
+                $found = false;
+                foreach ($filterList as $filters) {
+                    foreach ($filters as $filter) {
+                        if ($filter['field'] === $translatedField) {
+                            $found = true;
+                            break;
+                        }
+                    }
+                }
+                if (!$found) {
+                    $params->addFilter("$translatedField:$defaultValue");
+                }
+            }
+        }
+    }
+
+    /**
+     * Proxy a method call to parent class and all backend params classes
+     *
+     * @param string $method Method
+     * @param array  $params Method parameters
+     *
+     * @return mixed
+     */
+    protected function proxyMethod(string $method, array $params)
+    {
+        $result = call_user_func_array(['parent', $method], $params);
+        foreach ($this->searchParams as $searchParams) {
+            $result = call_user_func_array([$searchParams, $method], $params);
+        }
+        return $result;
+    }
+
+    /**
+     * Translate a facet
+     *
+     * @param string $field     Facet field
+     * @param string $backendId Backend ID
+     *
+     * @return string
+     */
+    protected function translateFacet(string $field, string $backendId): string
+    {
+        $prefix = '';
+        if (substr($field, 0, 1) === '~') {
+            $field = substr($field, 1);
+            $prefix = '~';
+        }
+        $fieldConfig = $this->mappings['Facets']['Fields'][$field] ?? [];
+        $mappings = $fieldConfig['Mappings'][$backendId] ?? [];
+        return !empty($mappings['Field']) ? ($prefix . $mappings['Field']) : '';
+    }
+
+    /**
+     * Check if the filter is a special Blender filter
+     *
+     * @param string $filter Filter
+     *
+     * @return bool
+     */
+    protected function isBlenderFilter(string $filter): bool
+    {
+        [$field] = $this->parseFilter($filter);
+        if (substr($field, 0, 1) === '~') {
```

Comment:
I hadn't found a use case for that as I was thinking that the filters would be used as checkbox filters, and NOT filter won't make sense there. But now that I think of it, it would certainly be possible to also support a `blender_backend` pseudo-facet that would display as a normal side facet, so adding NOT support would make sense.

Reply:
Now done.